Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS::Mac.os_name(ENV["HOMEBREW_OSX_VERSION"])
```

Comment:
@MikeMcQuaid How about I move `Class KegOnlyReason` in `formula_support` to `/extend/os/mac` (as we do for `formula_cellar_checks`) ? As the whole class is macOS-only and ideally we should not have it outside `Library/Homebrew/os/mac`, right? I am wondering how to specify that this method is macOnly and instruct the test should go into the macOs-only branch.